src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -240,6 +250,50 @@ private void transferRoutes(){
                     e.getIn().setBody(objectMapper.writeValueAsString(response));
                 });
 
+        //fetch transfer details based on client correaltion id
+        from("rest:GET:/channel/txnState/{correlationId}")
+                .id("transfer-details-correlationid")
+                .log(LoggingLevel.INFO, "## CHANNEL -> inbound transferDetail request for client correlation Id" +
+                        " ${header.correlationId}")
+                .process(e -> {
+                    String tenantId = e.getIn().getHeader("Platform-TenantId", String.class);
+                    if (tenantId == null || !dfspIds.contains(tenantId)) {
+                        throw new RuntimeException("Requested tenant " + tenantId + " not configured in the connector!");}
```

Comment:
do not use + here, use string format.
DFSP and tenants are not the same by design,  can we be more specific on this error?

Reply:
Currently our system designed in a way it expects tenant ids and dfspids to be same. The dfspid array is picked from application.yml that gets overwriiten by bb properties file.
<img width="555" alt="Screenshot 2022-12-23 at 6 17 42 PM" src="https://user-images.githubusercontent.com/92036361/209338864-d4152a9d-813c-43ab-a1ea-fc4bc9ff9ffa.png">


The logic is trying to check if the tenants coming in header exists in this DFSPID list or not, so I am not entirely sure what would be an ideal error message here

Reply:
do not use + to concatenate the message